File: dam/examen3/examen3/DataModel.swift
//
//  DataModel.swift
//  examen3
//
//  Created by roger on 11/04/25.
//

import Foundation

struct Pelicula: Identifiable, Equatable {
    let id: UUID
    var titulo: String
    var director: String
    var anio: Int
    var resena: String
    var calificacion: Int
    var vista: Bool
    
    init(id: UUID = UUID(), titulo: String, director: String, anio: Int, resena: String, calificacion: Int, vista: Bool) {
        self.id = id
        self.titulo = titulo
        self.director = director
        self.anio = anio
        self.resena = resena
        self.calificacion = calificacion
        self.vista = vista
    }

    static let interestellar = Pelicula(
        titulo: "Interstellar",
        director: "Christopher Nolan",
        anio: 2014,
        resena: "Viaje a estrellas",
        calificacion: 5,
        vista: true
    )
    static let orgullo = Pelicula(
        titulo: "Orgullo y Prejuicio",
        director: "Joe Wright",
        anio: 2005,
        resena: "Adaptación",
        calificacion: 4,
        vista: true
    )
    static let dune = Pelicula(
        titulo: "Dune",
        director: "Denis Villeneuve",
        anio: 2020,
        resena: "Dune",
        calificacion: 4,
        vista: false
    )

    static let ejemplos = [
        interestellar,
        orgullo,
        dune,
    ]
}

File: dam/peliculas-react/peliculas-react/DataModel.ts
// DataModel.ts
import 'react-native-get-random-values';
import { v4 as uuidv4 } from 'uuid'; // For generating IDs

interface Pelicula {
  id: string;
  titulo: string;
  director: string;
  anio: number;
  resena: string;
  calificacion: number;
  vista: boolean;
}

class PeliculaClass implements Pelicula {
  id: string;
  titulo: string;
  director: string;
  anio: number;
  resena: string;
  calificacion: number;
  vista: boolean;

  constructor(
    titulo: string,
    director: string,
    anio: number,
    resena: string,
    calificacion: number,
    vista: boolean,
    id: string = uuidv4()
  ) {
    this.id = id;
    this.titulo = titulo;
    this.director = director;
    this.anio = anio;
    this.resena = resena;
    this.calificacion = calificacion;
    this.vista = vista;
  }
}

const interestellar: Pelicula = new PeliculaClass(
  "Interstellar",
  "Christopher Nolan",
  2014,
  "Viaje a estrellas",
  5,
  true,
  uuidv4()
);

const orgullo: Pelicula = new PeliculaClass(
  "Orgullo y Prejuicio",
  "Joe Wright",
  2005,
  "Adaptación",
  4,
  true,
  uuidv4()
);

const dune: Pelicula = new PeliculaClass(
  "Dune",
  "Denis Villeneuve",
  2020,
  "Dune",
  4,
  false,
  uuidv4()
);

const ejemplos: Pelicula[] = [interestellar, orgullo, dune];

export { Pelicula, PeliculaClass, ejemplos };